src/vscode/TextDocument.ts

```diff
@@ -46,13 +46,14 @@ export class MockTextDocument implements vscode.TextDocument {
         const lineNumber = typeof line === 'number' ? line : line.line;
         const fullLineText = this._lines[lineNumber];
         const text = fullLineText.replace(/\r?\n/, '');
+        const firstNonWhitespaceCharacterIndex = /^(\s*)/.exec(text)![1].length
```

Comment:
How about using `trimStart`, removing the need for a RegExp?
```suggestion
        const firstNonWhitespaceCharacterIndex = text.length - text.trimStart();
```

Reply:
@Jason3S happy to do that, but before I do: code there is meant to match as closely as possible the VS code impl - I wouldn't have written the check that way either, but figured if it was possible to be obviously doing the same thing as them.

Having said that, still refer with `trimStart`? If so, I'll apply.

Reply:
@harpocrates,

I see. Thank you.

https://github.com/microsoft/vscode/blob/56222f3441914d033791e7e7a09b99423864751b/src/vs/workbench/api/common/extHostDocumentData.ts#L284-L291